service/src/main/kotlin/fi/espoo/evaka/daycare/DaycareQueries.kt

```diff
@@ -514,17 +514,26 @@ SELECT EXISTS(
         }
         .exactlyOne<Boolean>()
 
-private data class UnitOperationDays(val id: DaycareId, val operationDays: List<Int>)
+data class UnitOperationDays(
+    val operationDays: Set<DayOfWeek>,
+    val shiftCareOperationDays: Set<DayOfWeek>
+)
 
-fun Database.Read.getUnitOperationDays(): Map<DaycareId, Set<DayOfWeek>> =
-    createQuery { sql("""
-SELECT id, operation_days
+fun Database.Read.getUnitOperationDays(unitIds: Set<DaycareId>): Map<DaycareId, UnitOperationDays> =
+    createQuery {
+            sql(
+                """
+SELECT id, operation_days, shift_care_operation_days
 FROM daycare
-""") }
-        .mapTo<UnitOperationDays>()
-        .useIterable { rows ->
-            rows.fold(mutableMapOf()) { acc, row ->
-                acc[row.id] = row.operationDays.map { DayOfWeek.of(it) }.toSet()
-                acc
-            }
+WHERE daycare.id = ANY(${bind(unitIds)})
+"""
+            )
+        }
+        .map {
+            val operationDays = column<List<Int>>("operation_days").map { DayOfWeek.of(it) }.toSet()
+            val shiftCareOperationDays =
+                column<List<Int>?>("shift_care_operation_days")?.map { DayOfWeek.of(it) }?.toSet()
+            column<DaycareId>("id") to
+                UnitOperationDays(operationDays, shiftCareOperationDays ?: emptySet())
```

Comment:
should the default for null be `operationDays`? now effectiveOperationDays for children with shift care service need would be empty

Reply:
Good catch! Fixed by removing the whole query and using a more generic daycare query instead, and doing the fallback logic in the business logic.